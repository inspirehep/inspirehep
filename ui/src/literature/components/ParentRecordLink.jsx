import React from 'react';
import PropTypes from 'prop-types';
import { Map } from 'immutable';
import { Link } from 'react-router-dom';
import { getRecordIdFromRef } from '../../common/utils';
import { LITERATURE } from '../../common/routes';

function ParentRecordLink({ parentRecord }) {
  return (
    <div>
      <span>Part of </span>
      <Link
        to={`${LITERATURE}/${getRecordIdFromRef(
          parentRecord.getIn(['record', '$ref'])
        )}`}
      >
        {parentRecord.get('title')}
      </Link>
    </div>
  );
}

ParentRecordLink.propTypes = {
  parentRecord: PropTypes.instanceOf(Map).isRequired,
};

export default ParentRecordLink;
