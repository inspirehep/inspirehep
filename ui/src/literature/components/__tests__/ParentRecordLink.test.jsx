import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';

import ParentRecordLink from '../ParentRecordLink';

describe('ParentRecordLink', () => {
  it('renders with parent record', () => {
    const parentRecord = fromJS({
      title: 'A title of book',
      record: { $ref: 'http://localhost:5000/api/literature/1234' },
    });
    const wrapper = shallow(<ParentRecordLink parentRecord={parentRecord} />);
    expect(wrapper).toMatchSnapshot();
  });
});
