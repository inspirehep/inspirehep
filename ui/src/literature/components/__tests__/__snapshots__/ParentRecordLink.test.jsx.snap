// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParentRecordLink renders with parent record 1`] = `
<div>
  <span>
    Part of 
  </span>
  <Link
    to="/literature/1234"
  >
    A title of book
  </Link>
</div>
`;
